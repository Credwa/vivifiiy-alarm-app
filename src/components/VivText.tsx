import React from 'react';
import { StyleSheet, Text } from 'react-native';
import Colors from '@/constants/Colors';

interface FontStyle {
  fontFamily: string;
  fontSize: number;
}

export type FontName =
  | 'TitleBig1'
  | 'TitleBig2'
  | 'Title1'
  | 'Title2'
  | 'Title3'
  | 'Title4'
  | 'Title5'
  | 'Title6'
  | 'Headline'
  | 'Body'
  | 'Callout'
  | 'Subhead'
  | 'Footnote'
  | 'Caption'
  | 'CardSubtitle'
  | 'CardTitle';

interface TextProps {
  children?: React.ReactNode;
  color?: string;
  fontName?: FontName;
}

export type VivTextProps = TextProps & Text['props'];

function setFontStyle(size: number, family: string): FontStyle {
  const fontStyle: FontStyle = { fontSize: size, fontFamily: family };
  return fontStyle;
}

export default function VivText(props: VivTextProps) {
  let fontStyle: FontStyle;
  switch (props.fontName) {
    case 'TitleBig1':
      fontStyle = setFontStyle(84, 'Nunito_400Regular');
      break;
    case 'TitleBig2':
      fontStyle = setFontStyle(80, 'Nunito_400Regular');
      break;
    case 'Title1':
      fontStyle = setFontStyle(48, 'Nunito_400Regular');
      break;
    case 'Title2':
      fontStyle = setFontStyle(44, 'Nunito_400Regular');
      break;
    case 'Title3':
      fontStyle = setFontStyle(34, 'Nunito_400Regular');
      break;
    case 'Title4':
      fontStyle = setFontStyle(28, 'Nunito_400Regular');
      break;
    case 'Title5':
      fontStyle = setFontStyle(22, 'Nunito_400Regular');
      break;
    case 'Title6':
      fontStyle = setFontStyle(20, 'Nunito_400Regular');
      break;
    case 'Headline':
      fontStyle = setFontStyle(17, 'Nunito_600SemiBold');
      break;
    case 'Body':
      fontStyle = setFontStyle(17, 'Nunito_400Regular');
      break;
    case 'Callout':
      fontStyle = setFontStyle(16, 'Nunito_400Regular');
      break;
    case 'Subhead':
      fontStyle = setFontStyle(15, 'Nunito_400Regular');
      break;
    case 'Footnote':
      fontStyle = setFontStyle(13, 'Nunito_400Regular');
      break;
    case 'Caption':
      fontStyle = setFontStyle(12, 'Nunito_400Regular');
      break;
    default:
      fontStyle = setFontStyle(17, 'Nunito_400Regular');
  }
  return <Text style={[styles.text, fontStyle, { color: props.color }, props.style]}>{props.children}</Text>;
}

const styles = StyleSheet.create({
  text: {
    color: Colors.greyLight2
  }
});
