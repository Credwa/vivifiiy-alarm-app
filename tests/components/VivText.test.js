import * as React from 'react';
import renderer from 'react-test-renderer';

import VivText from '@/components/VivText';

describe('<VivText />', () => {
  it(`renders correctly`, () => {
    const tree = renderer.create(<VivText>Snapshot test!</VivText>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
