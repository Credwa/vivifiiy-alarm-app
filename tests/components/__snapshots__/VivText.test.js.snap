// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VivText /> renders correctly 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#869DB2",
      },
      Object {
        "fontFamily": "Nunito_400Regular",
        "fontSize": 17,
      },
      Object {
        "color": undefined,
      },
      undefined,
    ]
  }
>
  Snapshot test!
</Text>
`;

exports[`<VivText /> renders headline text correctly 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#869DB2",
      },
      Object {
        "fontFamily": "Nunito_600SemiBold",
        "fontSize": 17,
      },
      Object {
        "color": "#869DB2",
      },
      undefined,
    ]
  }
>
  Snapshot test!
</Text>
`;
